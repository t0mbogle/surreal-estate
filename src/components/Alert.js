import React from "react";
import PropTypes from "prop-types";

const Alert = ({ message, success }) => {
  if (!message) return null; // Conditional rendering to only show message if truthy

  return (
    // conditional class naming based on success result (.alert-success/.alert-error)
    <div className={`alert alert-${success ? "success" : "error"}`}>
      {message}
    </div>
  );
};

export default Alert;

Alert.defaultProps = {
  success: false,
};

Alert.propTypes = {
  message: PropTypes.string.isRequired,
  success: PropTypes.bool,
};
