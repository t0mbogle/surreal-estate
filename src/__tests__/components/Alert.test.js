import React from "react";
import { render } from "@testing-library/react";
import Alert from "../../components/Alert";

describe("Alert", () => {
  it("displays an error message", () => {
    const { getByText } = render(<Alert message="Error!" />);

    expect(getByText(/Error/).textContent).toBe("Error!");
  });
  it("displays a success message", () => {
    const { getByText } = render(<Alert message="Success!" success />);

    expect(getByText(/Success/).textContent).toBe("Success!");
  });
  it("does not render an error/success message if message prop is empty", () => {
    const { asFragment } = render(<Alert message="" />);
    const alert = asFragment();

    expect(alert).toMatchSnapshot();
  });
});
