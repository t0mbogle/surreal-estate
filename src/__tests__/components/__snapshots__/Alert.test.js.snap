// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert does not render an error/success message if message prop is empty 1`] = `<DocumentFragment />`;
